### PR TITLE
 [SDK][Bot-Azure] Add AzureQueueStorage component

### LINKED ISSUE #909: bot-azure, AzureQueueStorage
Port AzureQueueStorage from dotnet

### PR DESCRIPTION
**Pull request by VictorGrycuk**
Fixes #909 

## Description

Implements AzureQueueStorage class and its unit test. This PR also fixes a few disparities with C# present in different classes.

## Specific Changes

  - Add AzureQueueStorage class to Bot Azure library
  - Add unit test for AzureQueueStorage
  - Add ActivityEventNames class to Bot Schema library
  - Add missing locale assignation in [TestAdapter](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder/Adapters/TestAdapter.cs#L436)
  - Fix unnecessary lowercasing in [TestAdapter](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder/Adapters/TestAdapter.cs#L129-L130)
  - Fix missing assignation of service URL in [ConversationReference](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Schema/ConversationReference.cs#L103)'s `getContinuationActivity()` method
  - Fix the test `ConfirmPrompt_Locale_Override_ChoiceDefaults` of [ConfirmPromptLocTests](https://github.com/microsoft/botbuilder-dotnet/blob/main/tests/Microsoft.Bot.Builder.Dialogs.Tests/ConfirmPromptLocTests.cs#L231), by adding a missing assignation of the custom locale

## Testing

These implementations were successfully validated by running the build and unit tests

image

image

### CODE REVIEW

**Comment by VictorGrycuk** [comment]
Hi @tracyboehrer, the unit test for this implementation requires [Azure Storage Emulator](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator), just like the CosmosDbPartitionedStorage.

We found this [prep-test](https://github.com/microsoft/botbuilder-js/blob/main/prep-test.cmd#L12) in BotBuilder JS, similar to CosmosDbPartitionedStorage.

File: libraries/bot-azure/src/main/java/com/microsoft/bot/azure/queues/AzureQueueStorage.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                                                   @Nullable Duration timeToLive) {
+        return CompletableFuture.supplyAsync(() -> {
+            if (createQueueIfNotExists) {
+                // This is an optimization flag to check if the container creation call has been made.
+                // It is okay if this is called more than once.
+                createQueueIfNotExists = false;
+                try {
+                    queueClient.create();
+                } catch (QueueStorageException e) {
+                    e.printStackTrace();
```
Comment (LeeParrishMSFT) [inline libraries/bot-azure/src/main/java/com/microsoft/bot/azure/queues/AzureQueueStorage.java:72]: Is it acceptable for the code to continue at this point or should we return here?

Comment (VictorGrycuk) [reply libraries/bot-azure/src/main/java/com/microsoft/bot/azure/queues/AzureQueueStorage.java]: Hi @LeeParrishMSFT, we improved this behaviour to throw a `RuntimeException` like the [CosmosDbPartitionedStorage](https://github.com/microsoft/botbuilder-java/blob/main/libraries/bot-azure/src/main/java/com/microsoft/bot/azure/CosmosDbPartitionedStorage.java#L285-L289) does.

**Comment by tracyboehrer** [comment]
@Batta32 Can you change this, and the other Storage impl to follow the pattern in CosmosDbPartitionStorageTests.  I changed 'assertEmulator' to 'runIfEmulator'. 

The relevant code is:

```
    private boolean runIfEmulator() {
        if (!emulatorIsRunning) {
            System.out.println(NO_EMULATOR_MESSAGE);
            return false;
        }

        return true;
    }
```

Then (for example):

```
    @Test
    public void createObjectCosmosDBPartitionTest() {
        if (runIfEmulator()) {
            super.createObjectTest(storage);
        }
    }

```

**Comment by Batta32** [comment]
Sure @tracyboehrer, we will apply those changes to follow the pattern of `CosmosDbPartitionStorageTests` in this PR and in PR #1066.

**Comment by Batta32** [comment]
@tracyboehrer, we finished applying the requested change here and also, we merged `main` into this branch.

We will continue working on the PR #1066.

**Comment by VictorGrycuk** [comment]
@tracyboehrer we were able to reproduce the error 400 by [downgrading ](https://github.com/microsoft/botbuilder-java/blob/7e22ebd5533c4887e17f55ad7ff89494893d3288/libraries/bot-azure/pom.xml#L79)the [azure-storage-queue](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/storage/azure-storage-queue) to a previous version (**12.0.0**). We also tested from version 12.1.0 to 12.8.0, and the error is **not present** when using the Azure Storage Emulator **v5.10.0**.

The issue seems to be a mismatch between the version of the emulator and of the library.
Could you please validate the version of the Azure Storage Emulator running in the pipelines?

**Comment by tracyboehrer** [comment]
@VictorGrycuk OK.  I think we have this sorted out.  At least the PR build succeeded.  If you can update from main again I promise we'll get this merged.

**Comment by Batta32** [comment]
@tracyboehrer done! The build passed 🎉!

**Review by tracyboehrer — APPROVED** [review]

**Pull request merged**